### PR TITLE
fix(slider): reflect to old value on empty input value

### PR DESCRIPTION
**Pull request by prima-lseg**
## Description

The input reflects back to previous value when the value is empty in slider input.

Fixes # (issue)
[DME-18110](https://jira.refinitiv.com/browse/DME-18110)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/slider/elements/slider.ts [hunk 1]
```diff
@@ -897,11 +897,16 @@ export class Slider extends FormFieldElement {
       return;
     }
 
-    const { value, name } = event.target as NumberField;
+    const elInput = event.target as NumberField;
```
Comment (bualoy-napat) [inline packages/elements/src/slider/elements/slider.ts]:
```suggestion
    const inputEl = event.target as NumberField;
```

File: packages/elements/src/slider/elements/slider.ts [hunk 2]
```diff
@@ -897,11 +897,16 @@ export class Slider extends FormFieldElement {
...
+    const { value, name } = elInput;
```
Comment (bualoy-napat) [inline packages/elements/src/slider/elements/slider.ts]:
```suggestion
    const { value, name } = inputEl;
```

File: packages/elements/src/slider/elements/slider.ts [hunk 3]
```diff
@@ -897,11 +897,16 @@ export class Slider extends FormFieldElement {
...
 
     const currentData = name as SliderDataName;
     const previousData = `${name}Previous` as SliderPreviousDataName;
 
+    if (!value) {
+      elInput.value = this[currentData];
```
Comment (bualoy-napat) [inline packages/elements/src/slider/elements/slider.ts]:
```suggestion
      inputEl.value = this[currentData];
```

File: packages/elements/src/slider/elements/slider.ts [hunk 4]
```diff
@@ -1297,17 +1302,24 @@ export class Slider extends FormFieldElement {
     if (this.minNumber > this.maxNumber) {
       return false;
     }
-    // Check if value is in range
+
+    // Check if value exists and is in range
```
Comment (bualoy-napat) [inline packages/elements/src/slider/elements/slider.ts:1306]:
```suggestion
    // Check if value exists
```

File: packages/elements/src/slider/elements/slider.ts [hunk 5]
```diff
@@ -1297,17 +1302,24 @@ export class Slider extends FormFieldElement {
...
+    if (!value && isNaN(value)) {
+      return false;
+    }
+
```
Comment (bualoy-napat) [inline packages/elements/src/slider/elements/slider.ts:1310]:
```suggestion

    // Check if value is in range
```

**Comment by bualoy-napat** [comment]
LGTM.

**Review by wattachai-lseg — COMMENTED** [review]
Let's remove `event.preventDefault();` from line 915. [blur event](https://w3c.github.io/uievents/#event-types-list) is not cancelable i.e. can't be `preventDefault()`.

File: packages/elements/src/slider/elements/slider.ts [hunk 6]
```diff
@@ -897,11 +897,16 @@ export class Slider extends FormFieldElement {
...
+    const inputEl = event.target as NumberField;
+    const { value, name } = inputEl;
 
     const currentData = name as SliderDataName;
     const previousData = `${name}Previous` as SliderPreviousDataName;
 
+    if (!value) {
+      inputEl.value = this[currentData];
+    }
+
```
Comment (wsuwt) [inline packages/elements/src/slider/elements/slider.ts]: Suggesting a minor improvement kub. I think this improves a bit of overall code readability.

```ts
    if (value) {
      if (this[currentData] !== value) {
        this.updateNotifyProperty(currentData, value);
        this[previousData] = value;
      }
    } else {
      inputEl.value = this[currentData];
    }

```

Comment (wattachai-lseg) [reply packages/elements/src/slider/elements/slider.ts]: if we were to refactor this part I would suggest:
```typescript
    if (!value) {
      inputEl.value = this[currentData];
    } else if (this[currentData] !== value) {
      this.updateNotifyProperty(currentData, value);
      this[previousData] = value;
    }
```

Comment (wsuwt) [reply packages/elements/src/slider/elements/slider.ts]: Either way is definitely an improvement kub. I just prefer starting the statement with positivity if it's not a guard clause.

**Comment by wattachai-lseg** [comment]
LGTM

**Review by bualoy-napat — APPROVED** [review]

**Pull request merged**